subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidateTaskPropertiesIntegrationTest.groovy

```diff
@@ -387,6 +391,58 @@ class ValidateTaskPropertiesIntegrationTest extends AbstractIntegrationSpec {
         file("build/reports/task-properties/report.txt").text == ""
     }
 
+    @Unroll
+    def "report setters for mutable type #type"() {
+        @Language("JAVA") myTask = """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.tasks.InputFiles;
+
+            public class MyTask extends DefaultTask {
+                private ${type} sources;
+                
+                @InputFiles
+                public ${type} getSources() { return sources; } 
+                public void setSources(${type} sources) { this.sources = sources; } 
+            }
+        """
+        file("src/main/java/MyTask.java") << myTask
+
+        when:
+        fails "validateTaskProperties"
+
+        then:
+        file("build/reports/task-properties/report.txt").text == """
+            Warning: Type 'MyTask': 'sources' of type '${type.replaceAll("<.+>", "")}' has redundant setter method.
+        """.stripIndent().trim()
+
+        where:
+        type << [ConfigurableFileCollection.name, "${Property.name}<String>", RegularFileProperty.name]
```

Comment:
I still think we should have `Property<String>` here.
We also have `DirectoryProperty` that could be added as a test case.

Reply:
Since this list is used in two test cases, maybe it's worth extracting them into a constant.

Reply:
(Sorry, I missed that you need the FQCN for `Property`, I think it's okay to leave it like this then.)